packages/discord.js/src/client/Client.js

```diff
@@ -239,7 +239,7 @@ class Client extends BaseClient {
    * @returns {boolean}
    */
   isReady() {
-    return this.ws.status === Status.Ready;
+    return this.ws.destroyed === false && this.ws.status === Status.Ready;
```

Comment:
I don't think we do `=== false` unless there's also another value with a different meaning. Negating the value should be fine:
```suggestion
    return !this.ws.destroyed && this.ws.status === Status.Ready;
```